civiform/civiform
# Questions list view universal section and sorting

### Description

This makes five changes.

1. There was a bug where when you had a published (lifecycle stage ACTIVE) universal question, if you edited it and unset universal, it would not remove universal from the question until you unset it again.  This was due to how we were creating a new draft of a question, where we copy all of the QuestionTags from the old question directly.  We still have to do this since we don't maintain the state of the export tags in the QuestionDefinition (yet), but now we'll explicitly remove the universal tag if the new QuestionDefinition#isUniversal is false.
2. In the questions list view, universal questions are now in their own separate section.  If no questions are marked universal, it will look like it does today.  If one or more are, you will see them in their own section with a USWDS banner and a "all other questions" section for the rest.  This was in the mocks for the question bank but not the questions list view.  After discussing with @sijiayam , we decided the questions list view should have it as well.
3. Since we now have multiple sections we need to sort independently, this modifies the typescript code to sort all of the sections.
4. Moves the browser test to `civiform_admin_question_list` which is more where it belongs.
5. Adds a line to build.sbt to allow printing to stdout from inside tests. Normally, nothing would get printed when tests are run in forking mode (which is the default).  With this, we can `System.out.println` to our heart's desire.

## Release notes

The title of the pull request will be used as the default release notes description. If more detail is needed to communicate to partners the scope of the PR's changes, use this release notes section.

### Checklist

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)


#### User visible changes


It looks...not so great on mobile and 200% with elements overlapping, but that's unrelated to this PR.

#### New Features


### Instructions for manual testing

1. Ensure that removing universal from a published universal question removes it successfully the first time in the new draft.
2. Ensure the questions list view displays universal and other questions appropriately.
3. Ensure the questions list view sorts the question sections appropriately.

### Issue(s) this completes

https://github.com/civiform/civiform/issues/5972
